lambdas/service/app.py

```diff
@@ -356,6 +357,7 @@ def validate_facet(value):
 
 def validate_params(query_params: Mapping[str, str],
                     allow_extra_params: bool = False,
+                    mandatory_params: Iterable[str] = None,
```

Comment:
Instead of adding a new parameter, please consider this approach:

```
class Mandatory:

    def __init__(self, validator) -> None:
        super().__init__()
        self._validator = validator

    def __call__(self, value):
        return self._validator(value)
```

This could then be used as­ `validate_params(…,foo=Mandatory(int))` which I think reads better. You can then compute `mandatory_params` by taking the values in `validators` that are `instanceof(Mandatory)`.

Reply:
Okay. I like this approach.